packages/create-snap/src/cmds/init/initUtils.ts

```diff
@@ -74,30 +74,44 @@ export function isGitInstalled() {
  * @returns True if it's a git repository otherwise false.
  */
 export function isInGitRepository(directory: string) {
-  try {
-    spawnSync('git', ['rev-parse', '--is-inside-work-tree'], {
-      stdio: 'ignore',
-      cwd: pathUtils.resolve(__dirname, directory),
-    });
-    return true;
-  } catch (error) {
-    return false;
-  }
+  const result = spawnSync('git', ['rev-parse', '--is-inside-work-tree'], {
+    stdio: 'ignore',
+    cwd: pathUtils.resolve(__dirname, directory),
+  });
+
+  return result.status === 0;
 }
 
 /**
- * Init a git repository.
+ * Init a git repository and make the first commit.
  *
  * @param directory - The directory to init.
  */
-export function gitInit(directory: string) {
-  try {
-    spawnSync('git', ['init'], {
+export function gitInitWithCommit(directory: string) {
+  const commands = [
+    {
+      cmd: 'git',
+      params: ['init'],
+    },
+    {
+      cmd: 'git',
+      params: ['add', '.'],
+    },
+    {
+      cmd: 'git',
+      params: ['commit', '-m', 'Initial commit from create-snap.'],
```

Comment:
Commits usually don't end with a period.

```suggestion
      params: ['commit', '-m', 'Initial commit from create-snap'],
```

Maybe we can use the full package name? 

```suggestion
      params: ['commit', '-m', 'Initial commit from "@metamask/create-snap"'],
```